src/main/java/org/tikv/common/operation/KVErrorHandler.java

```diff
@@ -229,10 +229,14 @@ public boolean handleResponseError(BackOffer backOffer, RespT resp) {
         throw new StatusRuntimeException(Status.UNKNOWN.withDescription(error.toString()));
       }
 
-      logger.warn(String.format("Unknown error %s for region [%s]", error.toString(), ctxRegion));
+      logger.warn(String.format("Unknown error %s for region [%s]", error, ctxRegion));
       // For other errors, we only drop cache here.
       // Upper level may split this task.
       invalidateRegionStoreCache(ctxRegion);
+      // retry if raft proposal is dropped, it indicates the store is in the middle of transition
+      if (error.getMessage().contains("Raft Proposal Dropped")) {
+        return true;
```

Comment:
`invalidateRegionStoreCache` should be called in this case?

Reply:
it has been called above.

Reply:
i mean `invalidateRegionStoreCache` should not be called in this case.

Reply:
it should be called, because the region info is incorrect.